operator/v1/types_ingress.go

```diff
@@ -0,0 +1,119 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterIngress describes a discrete, managed ingress component for the
+// cluster. This includes a router, and may also include a load balancer and
+// external wildcard DNS records for that load balancer.
+type ClusterIngress struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ClusterIngressSpec   `json:"spec,omitempty"`
+	Status ClusterIngressStatus `json:"status,omitempty"`
+}
+
+// ClusterIngressSpec is the specification of the desired behavior of the
+// ClusterIngress.
+type ClusterIngressSpec struct {
+	// IngressDomain is a DNS name suffix serviced by the ClusterIngress. If
+	// specified, this value:
+	//
+	// * Is added automatically to the status of Routes to inform downstream systems.
+	// * May be used for a wildcard DNS alias to any provisioned load balancer.
+	// * May be used to generate a default wildcard certificate.
+	//
+	// If nil, the default value from the cluster ingress config is used.
+	//
+	// Once set, this field cannot be changed.
+	IngressDomain *string `json:"ingressDomain"`
+
+	// Replicas is the desired number of router instances.
+	Replicas int32 `json:"replicas"`
+
+	// HighAvailability describes the kind of high availability mechanism to apply
+	// to the cluster ingress. If nil, a default appropriate for the
+	// infrastructure platform is used.
+	//
+	// Once set, this field cannot be changed.
+	HighAvailability *ClusterIngressHighAvailability `json:"highAvailability"`
+
+	// DefaultCertificateSecret is the name of a secret containing the default
+	// certificate configuration. The secret is assumed to contain:
+	//
+	// * tls.crt
+	// * tls.key
+	DefaultCertificateSecret *string `json:"defaultCertificateSecret"`
```

Comment:
Should be `string` and empty means same as nil.

Reply:
We've talked and do think this should be a value type and use empty string as an indicator to use defaults -- currently empty string means "use the cert baked into the image" which we don't think is useful

Reply:
@smarterclayton, would using `
DefaultCertificate *corev1.LocalObjectReference` for this field be acceptable?

Reply:
I changed to a reference